src/drt/src/db/drObj/drNet.h

```diff
@@ -135,7 +137,27 @@ class drNet : public drBlockObject
     numMarkers_ = 0;
     routed_ = false;
   }
-  void setFrNet(frNet* in) { fNet_ = in; }
+  void updateFrNet(frNet* in)
+  {
+    fNet_ = in;
+    if (hasNDR())
+      maxRipupAvoids_ = NDR_NETS_RIPUP_HARDINESS;
+    if (isClockNetTrunk())
+      maxRipupAvoids_ = CLOCK_NETS_TRUNK_RIPUP_HARDINESS;
+    else if (isClockNetLeaf())
+      maxRipupAvoids_ = CLOCK_NETS_LEAF_RIPUP_HARDINESS;
```

Comment:
the hasNDR result will be overwritten for clocks so just make it an else if at the end

Reply:
I will actually make the lower assignments get the max value. NDRs should be less hard but this way things are more generic.